src/components/BottomNavigation/BottomNavigationBar.tsx

```diff
@@ -746,6 +763,7 @@ const BottomNavigationBar = ({
               centered: true,
               rippleColor: isV3 ? 'transparent' : touchColor,
               onPress: () => handleTabPress(index),
+              onLongPress: () => handleTabLongPress(index),
```

Comment:
Since `handleTabPress` and `handleTabLongPress` have the same logic to create an event object, can you make a function `eventForIndex`:

```ts
const eventForIndex = (index: number) => ({
   route: navigationState.routes[index],
   defaultPrevented: false,
   preventDefault: () => {
     event.defaultPrevented = true;
   },
});
```

Remove `handleTabPress` and `handleTabLongPress` and then your handlers can be:
```suggestion
              onPress: () => onTabPress(eventForIndex(index)),
              onLongPress: () => onTabLongPress?.(eventForIndex(index)),
```